src/kv/map.h

```diff
@@ -205,101 +199,40 @@ namespace kv
       }
     }
 
-    // Serialisation-related methods
-    void serialise(KvStoreSerialiser& s, bool include_reads) override
```

Comment:
`serialise` and `deserialised` moved from `TxView` to `Map`. They deal only with a `ChangeSet` (which could be owned and populated by anyone) and `Map` state. There are a few important caveats to this, where we move checks that were inside these methods to the caller (roughly speaking, the caller knows and decides "does this contain interesting things I want to serialise", and potentially doesn't call, rather than the method having an early-out).